### PR TITLE
You can now add and remove orders from RejectsOrders

### LINKED ISSUE #10978: Make RejectsOrders upgradable
Requested by @ABrandau.

### PR DESCRIPTION
**Pull request by SzabKel**
Closes #10978

### CODE REVIEW

File: OpenRA.Game/Traits/RejectsOrders.cs [hunk 1]
```diff
@@ -41,5 +41,17 @@ public static bool AcceptsOrder(this Actor self, string orderString)
 			var r = self.TraitOrDefault<RejectsOrders>();
 			return r == null || r.Except.Contains(orderString);
 		}
+
+		public static void AddOrder(this Actor self, string orderString)
+		{
+			var r = self.TraitOrDefault<RejectsOrders>();
+			r.Except.Add(orderString);
```
Comment (pchote) [inline OpenRA.Game/Traits/RejectsOrders.cs]: These will throw NullReferenceExceptions if the actor doesn't have RejectsOrders.

**Comment by pchote** [comment]
This doesn't appear to implement the upgrade support requested in #10978.

**Comment by SzabKel** [comment]
What does upgrading a reject order means?

**Comment by pchote** [comment]
It means implementing the `IUpgradable` / `UpgradableTrait` logic so that it can be toggled using the [actor upgrades](https://github.com/OpenRA/OpenRA/wiki/Actor-Upgrades) system.

**Comment by GraionDilach** [comment]
`RejectsOrders` already had a YAMl tag of filtering out orders even. I don't understand what this PR is supposed to do at the current stage.

**Comment by reaperrr** [comment]
@SzabKel Sorry, we can't take this as it is now so I'm closing this to clear the PR queue. If you re-implement it they way it was meant in #10978, feel free to update and reopen it.

**Pull request closed without merging**